v2/tools/generator/internal/codegen/pipeline/resource_registration_file.go

```diff
@@ -503,3 +470,103 @@ func (r *ResourceRegistrationFile) getImportedPackages() map[astmodel.PackageRef
 
 	return result
 }
+
+func (r *ResourceRegistrationFile) makeWatchesExpr(typeName astmodel.TypeName, codeGenerationContext *astmodel.CodeGenerationContext) dst.Expr {
+	secretWatchesExpr := r.makeSecretWatchesExpr(typeName, codeGenerationContext)
+
+	if secretWatchesExpr == nil {
+		return nil
+	}
+
+	sliceBuilder := astbuilder.NewSliceLiteralBuilder(astmodel.WatchRegistrationType.AsType(codeGenerationContext), true)
+	sliceBuilder.AddElement(secretWatchesExpr)
+	return sliceBuilder.Build()
+}
+
+func (r *ResourceRegistrationFile) makeSecretWatchesExpr(typeName astmodel.TypeName, codeGenerationContext *astmodel.CodeGenerationContext) dst.Expr {
+	secretKeys, ok := r.secretPropertyKeys[typeName]
+	if !ok {
+		return nil
+	}
+
+	if len(secretKeys) == 0 {
+		return nil
+	}
+
+	newWatchBuilder := astbuilder.NewCompositeLiteralBuilder(astmodel.WatchRegistrationType.AsType(codeGenerationContext))
+	sort.Strings(secretKeys)
+
+	// Not using astbuilder here because we want this to go onto a single line
+	source := &dst.CompositeLit{
+		Type: astmodel.ControllerRuntimeSourceKindType.AsType(codeGenerationContext),
+		Elts: []dst.Expr{
+			&dst.KeyValueExpr{
+				Key: dst.NewIdent("Type"),
+				Value: astbuilder.AddrOf(
+					&dst.CompositeLit{
+						Type: astmodel.SecretType.AsType(codeGenerationContext),
+					}),
+			},
+		},
+	}
+	newWatchBuilder.AddField("Src", astbuilder.AddrOf(source))
+
+	secretKeyParams := make([]dst.Expr, 0, len(secretKeys))
+	for _, secretKey := range secretKeys {
+		secretKeyParams = append(secretKeyParams, astbuilder.StringLiteral(secretKey))
+	}
+
+	listTypeName := typeName.WithName(typeName.Name() + "List").AsType(codeGenerationContext)
+
+	eventHandler := astbuilder.CallFunc(
+		"watchSecretsFactory",
+		astbuilder.SliceLiteral(dst.NewIdent("string"), secretKeyParams...),
+		astbuilder.AddrOf(astbuilder.NewCompositeLiteralBuilder(listTypeName).Build()))
+	newWatchBuilder.AddField("MakeEventHandler", eventHandler)
+
+	return newWatchBuilder.Build()
+}
+
+func (r *ResourceRegistrationFile) makeConfigMapWatchesExpr(typeName astmodel.TypeName, codeGenerationContext *astmodel.CodeGenerationContext) dst.Expr {
```

Comment:
Are we using this method anywhere as of now?

Reply:
This is used now